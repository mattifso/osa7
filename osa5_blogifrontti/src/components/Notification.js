import React from 'react'
import { connect } from 'react-redux'

export class Notification extends React.Component {
  render() {
    const notification = this.props.notification
    return (
      notification ?
        notification.isError ?
          <div className="errorMessage" style={{ border: 1, borderStyle: 'solid', borderColor: 'red', padding: 15, width: '25em' }}>
            {notification.message}
          </div> :
          <div className="successMessage" style={{ border: 1, borderStyle: 'solid', borderColor: 'green', padding: 15, width: '25em' }}>
            {notification.message}
          </div>
        : null
    )
  }
}

const mapStateToProps = (state) => {
  return {
    notification: state.notification,
  }
}

const ConnectedNotification = connect(mapStateToProps, null)(Notification)

export default ConnectedNotification