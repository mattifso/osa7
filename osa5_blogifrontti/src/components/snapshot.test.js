import React from 'react'
import { Notification } from './Notification'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

describe('Notification snapshot tests', () => {
  let notificationComponent
  beforeAll(() => {
    notificationComponent = shallow(<Notification />)
  })
  it('success renders correctly', () => {
    notificationComponent.setProps({ notification: { isError: false, message: 'hello' } })
    const tree = renderer
      .create(notificationComponent)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('error renders correctly', () => {
    notificationComponent.setProps({ notification: { isError: true, message: 'error' } })
    const tree = renderer
      .create(notificationComponent)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})