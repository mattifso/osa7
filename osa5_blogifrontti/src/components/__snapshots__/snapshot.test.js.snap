// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification snapshot tests error renders correctly 1`] = `
<div
  className="errorMessage"
  style={
    Object {
      "border": 1,
      "borderColor": "red",
      "borderStyle": "solid",
      "padding": 15,
      "width": "25em",
    }
  }
>
  error
</div>
`;

exports[`Notification snapshot tests success renders correctly 1`] = `
<div
  className="successMessage"
  style={
    Object {
      "border": 1,
      "borderColor": "green",
      "borderStyle": "solid",
      "padding": 15,
      "width": "25em",
    }
  }
>
  hello
</div>
`;
